### PR TITLE
feat: express server

### PR DESCRIPTION
**Pull request by JustD1d1t**

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const express = require("express");
+const app = express();
+
+//init database
```
Comment (MaciekPon) [inline app/app.js:4]: why are you leaving these comments here?

Comment (JustD1d1t) [reply app/app.js]: Removed

**Comment by MaciekPon** [comment]
LGTM

File: app/app.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+const express = require("express");
+const app = express();
```
Comment (ruljin) [inline app/app.js:2]: I don't see any profit in dividing it into modules. For me, it should be merged with `index.js`

File: app/app.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+const express = require("express");
```
Comment (ruljin) [inline app/app.js:1]: Why not use `ES6` and use `import` and `export`?

File: app/index.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const app = require("./app");
+const port = 3000;
```
Comment (ruljin) [inline app/index.js:2]: `process.env.PORT || 3000`

File: app/index.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+
+app.listen(port, () => {
```
Comment (ruljin) [inline app/index.js:4]: What if there will be an error while starting the server?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    "url": "https://github.com/CodersCampLittleDevs/CodersCamp2021-ProjectServerSideJavaScript-PetsAdoptAndCarePortal/issues"
+  },
+  "homepage": "https://github.com/CodersCampLittleDevs/CodersCamp2021-ProjectServerSideJavaScript-PetsAdoptAndCarePortal#readme",
+  "dependencies": {
+    "bcrypt": "^5.0.1",
+    "dotenv": "^16.0.0",
+    "express": "^4.17.3",
+    "mongodb": "^4.4.0",
+    "mongoose": "^6.2.2",
+    "nodemon": "^2.0.15"
```
Comment (ruljin) [inline package.json:27]: It should be ` dev` dependency

**Review by ruljin — CHANGES_REQUESTED** [review]
Changes requested

File: app/app.js (hunk 2 above)
Comment (JustD1d1t) [reply app/app.js:2]: merged app.js with index.js

File: app/app.js (hunk 3 above)
Comment (JustD1d1t) [reply app/app.js:1]: My mistake

File: app/index.js (hunk 4 above)
Comment (JustD1d1t) [reply app/index.js:2]: I think that it should be implemented by person who has task with dotenv but okay, i'll do it as you wrote

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,2 @@
+/node_modules
```
Comment (ruljin) [inline .gitignore:1]: I think that this `/` is not needed here

File: app/index.js (hunk 5 above)
Comment (JustD1d1t) [reply app/index.js]: add handling error

File: package.json (hunk 6 above)
Comment (JustD1d1t) [reply package.json:29]: moved to dev

File: app/index.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import express from "express";
+const port = process.env.PORT || 3000;
+const app = express();
+
+app.listen(port, (error) => {
```
Comment (ruljin) [inline app/index.js:5]: Totally wrong as for me.
`server.listen(PORT, () => {console.log("Listening...!");}).on("error", (err) => {throw Error("MUUUUUUUUUU", err);});`

Comment (ruljin) [reply app/index.js:5]: Besides, why not set headers here?

**Review by ruljin — CHANGES_REQUESTED** [review]
Changes requested

File: .gitignore (hunk 7 above)
Comment (JustD1d1t) [reply .gitignore:1]: It also works

File: app/index.js (hunk 8 above)
Comment (JustD1d1t) [reply app/index.js]: Change handling and also add headers

File: package.json [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    "bcrypt": "^5.0.1",
+    "dotenv": "^16.0.0",
+    "express": "^4.17.3",
+    "mongodb": "^4.4.0",
+    "mongoose": "^6.2.2"
+  },
+  "devDependencies": {
+    "nodemon": "^2.0.15"
+  }
+}
```
Comment (DawidZlobecki) [inline package.json:31]: LGTM

File: app/index.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import express from "express";
+const port = process.env.PORT || 3000;
+const app = express();
+
+app.use(function (req, res, next) {
+  res.setHeader("Access-Control-Allow-Origin", ["*"]);
```
Comment (ruljin) [inline app/index.js:6]: From what I'm aware, just `'*'` is enough. You don't need to write it in an array.

File: app/index.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+  res.setHeader("Access-Control-Allow-Methods", "GET, POST, DELETE");
+  res.setHeader("Access-Control-Allow-Headers", "Content-Type");
+  next();
+});
+
+app
+  .listen(port, () => {
+    console.log(`Serwer uruchomiony na porcie ${port}`);
```
Comment (ruljin) [inline app/index.js:14]: English please

File: app/index.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+  })
+  .on("error", (err) => {
+    console.error("Wystąpił błąd przy uruchomieniu serwera " + err);
```
Comment (ruljin) [inline app/index.js:17]: English please

**Review by ruljin — CHANGES_REQUESTED** [review]
Chamges requested

**Review by ruljin — APPROVED** [review]
LGTM

**Pull request merged**